lib/LocalVerifiableCredentialStore.js

```diff
@@ -208,31 +209,19 @@ export class LocalVerifiableCredentialStore {
     const query = Array.isArray(credentialQuery) ? credentialQuery :
       [credentialQuery];
 
-    const _query = async ({example}) => {
-      const {type, trustedIssuer = []} = example;
+    const _query = async ({example, trustedIssuer = []}) => {
+      const {type} = example;
       // normalize trusted issuers to be an array
       const trustedIssuers = Array.isArray(trustedIssuer) ? trustedIssuer :
         [trustedIssuer];
 
-      // build query to find all VCs that match any combination of type+issuer
-      const query = [];
       const issuers = trustedIssuers.filter(({required}) => required)
         .map(({issuer}) => issuer);
       const types = Array.isArray(type) ? type : [type];
-      for(const type of types) {
-        if(issuers.length === 0) {
-          query.push({type});
-          continue;
-        }
-        for(const issuer of issuers) {
-          query.push({type, issuer});
-        }
-      }
-      const queryType = query.map(q => q.type);
-      const queryTypes = Array.isArray(queryType) ? queryType : [queryType];
       const selector = {
         'meta.profileId': profileId,
-        'content.type': {$in: queryTypes}
+        'content.type': {$in: types},
+        'meta.issuer': {$in: issuers}
```

Comment:
Since there can't be more than one issuer per VC -- I would expect an `$or` query to be involved here. I'd expect there to be an `$or` array (if that syntax is supported here) with one entry per issuer with the profile ID and types matched up against that issuer. Right?

Reply:
@dlongley I am not sure if I'm using the $or operator correctly here 3c52566. With this change, CHAPI is able to find the credential when using a matching issuer, and is unable to find any credentials when a non-matching issuer is used or when no trustedIssuer is passed. Not sure if that is the behavior we want here.